lib/cli/src/Cardano/CLI.hs

```diff
@@ -630,38 +638,92 @@ cmdWalletUpdateName mkClient =
             (ApiT wId)
             (WalletPutData $ Just (ApiT wName))
 
+data UpdatePassphraseCredential = MnemonicCredentials | OldPasswordCredentials
+    deriving Eq
+
+-- | Which json schema to use for output
+useMnemonicOption :: Parser UpdatePassphraseCredential
+useMnemonicOption = flag OldPasswordCredentials MnemonicCredentials
+    $ long "mnemonic"
+        <> help "use mnemonic to authorize the passphrase change"
+
 -- | Arguments for 'wallet update passphrase' command
 data WalletUpdatePassphraseArgs = WalletUpdatePassphraseArgs
     { _port :: Port "Wallet"
     , _id :: WalletId
+    , _mnemonic :: UpdatePassphraseCredential
     }
 
-cmdWalletUpdatePassphrase
-    :: ToJSON wallet
-    => WalletClient wallet
-    -> Mod CommandFields (IO ())
-cmdWalletUpdatePassphrase mkClient =
-    command "passphrase" $ info (helper <*> cmd) $ mempty
-        <> progDesc "Update a wallet's passphrase."
-  where
-    cmd = fmap exec $ WalletUpdatePassphraseArgs
-        <$> portOption
-        <*> walletIdArgument
-    exec (WalletUpdatePassphraseArgs wPort wId) = do
-        res <- sendRequest wPort $ getWallet mkClient $ ApiT wId
-        case res of
-            Right _ -> do
-                wPassphraseOld <- getPassphrase
-                    "Please enter your current passphrase: "
-                wPassphraseNew <- getPassphraseWithConfirm
-                    "Please enter a new passphrase: "
-                runClient wPort (const mempty) $
-                    putWalletPassphrase mkClient (ApiT wId) $
-                        WalletPutPassphraseData
-                            (ApiT wPassphraseOld)
-                            (ApiT wPassphraseNew)
-            Left _ ->
-                handleResponse Aeson.encodePretty res
+class CmdWalletUpdatePassphrase wallet where 
+    cmdWalletUpdatePassphrase
+        :: ToJSON wallet
+        => WalletClient wallet
+        -> Mod CommandFields (IO ())
+instance CmdWalletUpdatePassphrase ApiWallet where
+    cmdWalletUpdatePassphrase mkClient =
```

Comment:
The prefix `mk` in the `mkClient` suggest that its a constructor-like function. I'd name it simply `client` or `walletClient` instead.

Reply:
I agree on the drive but I will not change this code as I'm not the original author. I understand the idea of making code better, but better is relative and we should use code reviews for this kind of change. IMO once in production code should be changed as far as the PR needs the changes, or alternatively, we could make a specific PR for cleaning up consistently a module or functionality. In this specific case, we have 2 functions with the same hairy names, and they should be both fixed for consistency, but you see how easy this can go far out of PR focus.